asset/src/elasticsearch_sender_api/schema.ts

```diff
@@ -34,16 +36,32 @@ export default class Schema extends ConvictSchema<ElasticsearchSenderConfig> {
             return apiName === DEFAULT_API_NAME || apiName.startsWith(`${DEFAULT_API_NAME}:`);
         });
 
+        const { connectors } = this.context.sysconfig.terafoundation;
+
+        // hack to get around default connection check until schema updates for routed_sender
+        // replaces default connection with routed sender connection
+        if (connectors.elasticsearch.default == null && getOpConfig(job, 'routed_sender')) {
+            this._applyRoutedSenderConnection(job, apiConfigs);
+        }
+
         apiConfigs.forEach((apiConfig: AnyObject) => {
             const { connection } = apiConfig;
 
-            const { connectors } = this.context.sysconfig.terafoundation;
             const endpointConfig = connectors.elasticsearch[connection];
 
             if (endpointConfig == null) throw new Error(`Could not find elasticsearch endpoint configuration for connection ${connection}`);
         });
     }
 
+    _applyRoutedSenderConnection(job: ValidatedJobConfig, apiConfigs: APIConfig[]): void {
```

Comment:
This really needs an explanation of why it is here. Having a dependency like this is really bad so it's a complete hack and should be labeled as such at a minimum.

Reply:
added comment, and hopefully this combined with the comment on line 41 explains why this is here